lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -9429,22 +9396,6 @@ void Emit(ParseNode *pnode, ByteCodeGenerator *byteCodeGenerator, FuncInfo *func
         return;
     }
 
-    if (EmitUseBeforeDeclaration(pnode, byteCodeGenerator, funcInfo))
```

Comment:
Oh yeah.  I've been wanting to make this change for a long while.

I had a change that did it part way from years ago but I couldn't find the pack.  I think the work I did here to redo it resulted in a cleaner change though.

---
In reply to: [62540552](https://github.com/Microsoft/ChakraCore/pull/936#discussion_r62540552) [](ancestors = 62540552)